### PR TITLE
feat: add send events and API endpoint for uploading submissions to S3

### PR DESCRIPTION
**Pull request by jessicamcinchak**
These are the backend changes following on from #2999:
- Adds API endpoint `/upload-submisison/:localAuthority` within Send module to generate digital planning payload JSON & upload as a private file to a specified/static S3 folder within our main user data bucket
- Hooks up Hasura send events for this destination

**Testing:**
- Go to a supported submission service flow in Barnet's editor workspace (eg Prior Approval) 
  - Update Send component to select S3 option and publish
- Open `/published` link and fill out application (note that Pay is not enabled for Barnet and will show error, but you can "Continue" through without paying for this test and still generate a valid payload)
- After reaching Confirmation page, check Hasura events for successful audit log 
  - The success message will include a link the uploaded submission on S3 like below
  - Confirm that JSON file is private and you can download successfully if authenticated
  - The `lowcal_session` database record should also be marked as submitted

Screenshot from 2024-04-11 09-43-39

### CODE REVIEW

**Comment by jessicamcinchak** [comment]
[Automatically generate application forms/reports from ODP Schema JSON file using AWS S3, MS Power Automate & PowerBI](https://trello.com/c/FtJ2svyh/2861-automatically-generate-application-forms-reports-from-odp-schema-json-file-using-aws-s3-ms-power-automate-powerbi)

File: api.planx.uk/modules/file/service/uploadFile.ts [hunk 1]
```diff
@@ -59,20 +61,21 @@ const buildFileUrl = (key: string, path: "public" | "private") => {
...
+  const key = `${filekey || nanoid()}/${filename}`;
 
   const params = {
     ACL: process.env.AWS_S3_ACL,
     Key: key,
-    Body: file.buffer,
+    Body: file?.buffer || JSON.stringify(file),
     ContentDisposition: `inline;filename="${filename}"`,
-    ContentType: file.mimetype,
+    ContentType: file?.mimetype || "application/json",
```
Comment (jessicamcinchak) [inline api.planx.uk/modules/file/service/uploadFile.ts:78]: A bit surprisingly, `exportData` (our `DigitalPlanningApplication` typed object) satisfies type `Express.Multer.File` expected by these functions. It won't have a `buffer` & `mimetype` properties, but the `S3.PutObjectRequest.Body` accepts any string so I've just defined this simple `JSON.stringify()` fallback here.

Would be a good one to more intentionally revisit in a follow-up PR if this grows beyond a proof-of-concept!

**Review by DafyddLlyr — COMMENTED** [review]
Code-only review looks great! Going onto manual testing now (thanks for notes!) and I'll follow up shortly 👍

**Review by DafyddLlyr — APPROVED** [review]
- [x] The success message will include a link the uploaded submission on S3 like below
- [x] Confirm that JSON file is private and you can download successfully if authenticated
- [x] The lowcal_session database record should also be marked as submitted

Tested and working as expected! ✅

File: editor.planx.uk/src/@planx/components/Send/Editor.tsx [hunk 2]
```diff
@@ -65,62 +65,66 @@ const SendComponent: React.FC<Props> = (props) => {
...
+      if (
+        value === Destination.S3 &&
+        newCheckedValues.includes(value) &&
+        teamSlug !== "barnet"
+      ) {
+        alert(
+          "AWS S3 uploads are currently being prototyped with Barnet only. Please do not select this option for other councils yet.",
+        );
+      }
+    };
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply editor.planx.uk/src/@planx/components/Send/Editor.tsx:127]: Thanks for these validation reminders - will address all in followup PRs - going to merge this initial work so I can get test  links over to Kev sooner than later :+1:

**Pull request merged**